File: 01_Array/Q19_58. Length of Last Word/solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.length();
        int idx = n - 1;
        int wordLength = 0;

        while(idx >= 0 && s[idx] == ' ') idx--;

        while(idx >= 0 && s[idx] != ' '){
            idx--;
            wordLength++;
        }

        return wordLength;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(1)

File: 01_Array/Q19_58. Length of Last Word/solution.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        idx = n - 1
        word_length = 0

        while idx >= 0 and s[idx] == ' ':
            idx -= 1

        while idx >= 0 and s[idx] != ' ':
            idx -= 1
            word_length += 1

        return word_length


# Time Complexity -> O(n)
# Space Complexity -> O(1)
